src/utils/streams/__tests__/promise_from_streams.js

```diff
@@ -93,10 +93,8 @@ describe('promiseFromStreams', () => {
           },
 
           write(chunk, enc, cb) {
-            setTimeout(() => {
```

Comment:
Had to remove timeout since `write` would stop getting called after the pipeline finished. (caused by `read` pushing `null`)

Reply:
I think I see what you mean, based on the way I wrote this stream I'm guessing I assumed that `write()` would always be called when something was written to the duplex, but `stream.pipeline()` is now tearing down the writable side of the duplex when the readable side pushes `null`. Interesting... Shouldn't be an issue for anything we have in use right now.

Maybe we can update this implementation to actually correlate the data being written and the data it makes available for reading, maybe a duplex stream that works sort of like [`pairwise`](https://rxjs-dev.firebaseapp.com/api/operators/pairwise) but is delayed by 50ms or something...

Reply:
Good idea, I updated the test to have the 50ms timeout again and to push at the same time write callback is called.